src/main/scala/julienrf/json/derived/FieldAdapter.scala

```diff
@@ -0,0 +1,25 @@
+package julienrf.json.derived
+
+trait Adapter extends (String => String)
```

Comment:
Can you please add some scaladoc explaining the purpose of this class? Also, do you mind renaming it to `NameAdapter`?